src/platforms/common/blackpill-f4/usbdfu.c

```diff
@@ -75,4 +80,46 @@ int main(void)
 
 void dfu_event(void)
 {
+	static bool idle_state = false;
+	/* Ask systick to pause blinking for 1 second */
+	dfu_activity_counter = 10;
+	/* Blink it ourself */
+	SET_IDLE_STATE(idle_state);
+	idle_state = !idle_state;
+}
+
+static void sys_tick_init(void)
+{
+	/* Use SysTick at 10Hz to toggle-blink the blue LED at 5 Hz */
+	systick_set_clocksource(STK_CSR_CLKSOURCE_AHB_DIV8);
+	systick_set_reload(rcc_ahb_frequency / 8 / 10);
+	/* SYSTICK_IRQ with low priority */
+	nvic_set_priority(NVIC_SYSTICK_IRQ, 14U << 4U);
+	systick_interrupt_enable();
+	/* Start the heartbeat timer */
+	systick_counter_enable();
+}
+
+void sys_tick_handler(void)
+{
+	static int count = 0;
```

Comment:
Given the switch-case below, please make this at least unsigned - preferably a fixed width type like `uint8_t` or something like `size_t`. This should give you better codegen and prevent any possible undefined things happening with those `count--;`.

Reply:
Item accepted.
For this feature I was looking at your native hardware platform-specific code, namely https://github.com/blackmagic-debug/blackmagic/blob/main/src/platforms/native/usbdfu.c -- is that not good enough or outdated? I am often faced with review items on (mostly) copied code existing in adjacent platforms or drivers.

Reply:
That's outdated and needs the same update for correctness done - that probably means all of the usbdfu.c's need the same check done on them.